openem-ingestor/core/datasetingestor.go

```diff
@@ -0,0 +1,109 @@
+package core
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/paulscherrerinstitute/scicat/datasetIngestor"
+)
+
+func IngestDataset(
```

Comment:
There seems to be a lot of bits in `IngestDataset` that could be moved into the datasetIngestor package. I feel like core/datasetingestor should focus on executing the task. The fact that it has to run lots of different functions (CheckMetadata, UpdateMetaData, Upload*, MarkFIlesReady) seems to indicate a poorly designed base library. Should we fix that rather than duplicating code here and in `cmd`?